src/Appwrite/Extend/Exception.php

```diff
@@ -0,0 +1,186 @@
+    const COLLECTION_LIMIT_EXCEEDED  = 'collection_limit_exceeded';
+    const COLLECTION_DELETION_FAILED = 'collection_deletion_failed';
+    
+    /** Documents */
+    const DOCUMENT_NOT_FOUND         = 'document_not_found';
+    const DOCUMENT_INVALID_STRUCTURE = 'document_invalid_structure';
+    const DOCUMENT_MISSING_PAYLOAD   = 'document_missing_payload';
+    const DOCUMENT_ALREADY_EXISTS    = 'document_already_exists';
+
+    /** Attribute */
+    const ATTRIBUTE_NOT_FOUND           = 'attribute_not_found';
+    const ATTRIBUTE_UNKNOWN             = 'attribute_unknown';
+    const ATTRIBUTE_NOT_AVAILABLE       = 'attribute_not_available';
+    const ATTRIBUTE_FORMAT_UNSUPPORTED  = 'attribute_format_unsupported';
+    const ATTRIBUTE_DEFAULT_UNSUPPORTED = 'attribute_default_unsupported';
+    const ATTRIBUTE_ALREADY_EXISTS      = 'attribute_already_exists';
+    const ATTRIBUTE_LIMIT_EXCEEDED      = 'attribute_limit_exceeded';
+    const ATTRIBUTE_VALUE_INVALID       = 'attribute_value_invalid';
+
+    /** Indexes */
+    const INDEX_NOT_FOUND       = 'index_not_found';
+    const INDEX_LIMIT_EXCEEDED  = 'index_limit_exceeded';
+    const INDEX_ALREADY_EXISTS  = 'index_already_exists';
+
+    /** Query limit exceeded */
+    const QUERY_LIMIT_EXCEEDED = 'query_limit_exceeded';
+    const QUERY_INVALID        = 'query_invalid';
+
+    /** Projects */
+    const PROJECT_NOT_FOUND       = 'project_not_found';
+    const PROJECT_UNKNOWN         = 'project_unknown';
+
+    /** Webhooks */
+    const WEBHOOK_NOT_FOUND       = 'webhook_not_found';
+
+    /** Keys */
+    const KEY_NOT_FOUND       = 'key_not_found';
+
+    /** Platform */
+    const PLATFORM_NOT_FOUND       = 'platform_not_found';
+
+    /** Domain */
+    const DOMAIN_NOT_FOUND           = 'domain_not_found';
+    const DOMAIN_ALREADY_EXISTS      = 'domain_already_exists';
+    const DOMAIN_UNREACHABLE         = 'domain_unreachable';
+    const DOMAIN_VERIFICATION_FAILED = 'domain_verification_failed';
+
+    /** General */
+    const DEFAULT                   = 'default';
+    const ACCESS_FORBIDDEN          = 'access_forbidden';
+    const UNKNOWN_ORIGIN            = 'unknown_origin';
+    const SERVICE_DISABLED          = 'service_disabled';
+    const INVALID_READ_PERMISSIONS  = 'invalid_read_permissions';
+    const INVALID_WRITE_PERMISSIONS = 'invalid_write_permissions';
+    const UNAUTHORIZED_SCOPE        = 'unauthorized_scope';
+    const RATE_LIMIT_EXCEEDED       = 'rate_limit_exceeded';
+    const SMTP_DISABLED             = 'smtp_disabled';
+    const DATABASE_NOT_AVAILABLE    = 'database_not_available';
+    const CACHE_NOT_AVAILABLE       = 'cache_not_available';
+    const TIME_GAPS_DETECTED        = 'time_gaps_detected';
+    const STORAGE_NOT_READABLE      = 'storage_not_readable';
+    const STORAGE_NOT_WRITABLE      = 'storage_not_writable';
+    const ANTIVIRUS_NOT_AVAILABLE   = 'antivirus_not_available';
```

Comment:
Server errors (500) can be completely ignored.

Reply:
done 👍